xcode.mk

```diff
@@ -7,10 +7,19 @@ CMAKE_BOOST_OPTIONS = -DBoost_NO_BOOST_CMAKE=TRUE \
 	-DBOOST_ROOT="$(BOOST_ROOT)"
 endif
 
+# https://cmake.org/cmake/help/latest/envvar/MACOSX_DEPLOYMENT_TARGET.html
+export MACOSX_DEPLOYMENT_TARGET = 10.9
+
+CMAKE_OSX_OPTIONS = -DCMAKE_OSX_SYSROOT="$(shell xcrun --sdk macosx --show-sdk-path)" \
```

Comment:
How this option work for the build tools also puzzles me.
I struggled for quite a long time to configure the universal build of librime on Travis CI.
Though I succeeded a universal build much more easily on my local machine.

One of the must-have compiler flag for [building boost](https://github.com/rime/librime/commit/73da07e09f28f6ac486ecd4335d5fe47af4e134f) as universal binary is 
`-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk`
Somehow the default sdk desn't work. But it seem only necessary for Boot::Build toolchain; on the other hand, when I built librime with CMake everything worked out of the box.

So, my question: does `xcrun --sdk macosx --show-sdk-path` reliably output the correct sdk path, as the one I specified for building boost? In what circumstances is this flag required? I figured that it is necessary to specify an option explicitly if the default doesn't work, but what exactly is the non-working case?

Reply:
On my Mac, sometime cmake get a wrong path of SDK, usually caused by the update of SDK verison.

So I add this option to make sure cmake can get the right SDK path every build.

Reply:
With this, no need to do a full clean. https://gitlab.kitware.com/cmake/cmake/-/issues/19755

Reply:
I hope I now understand correctly that it's for rebuilding after an Xcode SDK upgrade, and the generated makefiles still refer to an invalid SDK path because there is no way for cmake to know the cached SDK path needs update.

That's an acceptable reason to have the option. Though personally I don't like bother with preventing such special cases to go wrong by specifying a distracting long argument every time. If I ran into it I'd rather do a manual `rm build/CMakeCache.txt`.